src/behaviours/draggable.js

```diff
@@ -296,11 +297,13 @@ export default function draggable(WrappedComponent) {
     animate: false,
     onDropped: () => {},
     onMove: () => {},
+    originIds: {},
   };
 
-  function mapStateToProps(state) {
+  function mapStateToProps(state, props) {
     return {
       zones: state.droppable.zones,
+      originIds: { promptId: props.promptId, stageId: props.stageId },
```

Comment:
Does this mean that another draggable on the same screen would be considered the same origin? Suggest we either manually provide a unique id, or we could auto generate an id on creation using `uniqueId` in lodash?

Reply:
This is trying to keep track of which stage and prompt the draggable was created/originated on. This is to help distinguish between where nodes were created, so only nodes created on other prompts and pulled in from the side panel can be dragged back to the side panel. Nodes created on the current prompt need to be deleted, not dragged to a side panel.

Reply:
That should probably be handled elsewhere, since any component (not just nodes) can be a draggable(). Would a uid enable us to do that, or do we need to have a think about how this might work?

Reply:
What would a uid change? Wouldn't it still need to be connected to the promptid/stageid somewhere along the lines, whether it comes in as a prop or not?

Reply:
Oh! I think I'm missing something here?

Can't the draggable make this distinction in it's onDrop method? as the originator of the Node, doesn't it already have this information about what got dropped?

If that's not the case, maybe we should think about adding some kind of `meta` prop which is associated with draggable/droppables that can contain arbitrary data.

Reply:
I'll have to think about that. I know there are three places we need this information -- we need to know whether to show the droppable area when we start dragging, we need to know whether to show a placeholder node when hovering, and we need to know whether to allow a node to be dropped. And it varies by combination of node and prompt, because nodes could come from different prompts originally.